### PR TITLE
Feature/add wrr

### PR DESCRIPTION
**Pull request by aleerizw**
Added an environment varaible APP_WEIGHT which determines the weight the app in the weighted round robin.
The weight is added as a tag to the service which is then parsed inside the haproxy template to set the weight

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -242,6 +242,7 @@ The group of variables that define application configuration.
 |APP_HEALTH_TTL     |  NO      | 25      | The time-to-live in seconds of a successful health check |
 |APP_HEALTH_INTERVAL|  NO      | 5       | Is the time in seconds between health checks |
 |APP_HEALTH_TIMEOUT |  No      | 5       | Is a value to wait before forcibly killing the health check exec (see the golang ParseDuration docs for this format |
+|APP_WEIGHT         |  No      |         | Set the weight for the application for HAProxy backends |
```
Comment (ahelal) [inline README.md]: should have default value

File: etc/containerpilot.json5 [hunk 2]
```diff
@@ -95,9 +95,7 @@
         timeout: {{ .APP_HEALTH_TIMEOUT | default 5 }},
       },
       {{ end -}}
-      {{ if .APP_TAGS -}}
-      tags: [ {{ range $i := .APP_TAGS | split "," }}"{{ $i }}", {{ end }} ],
-      {{ end -}}
+      tags: [ {{ if .APP_WEIGHT -}}"WEIGHT:{{ .APP_WEIGHT }}",{{ end -}} {{ if .APP_TAGS -}}{{ range $i := .APP_TAGS | split "," }}"{{ $i }}", {{ end }} {{ end -}}],
```
Comment (ahelal) [inline etc/containerpilot.json5]: is empty tags tested ?

File: etc/haproxy.ctmpl [hunk 3]
```diff
@@ -59,7 +59,14 @@ backend {{ $ServiceName }}
     balance {{ $ServiceBalance }}
     option  {{ $ServiceMode }}log
     {{ range service ( printf "%s" $ServiceName) }}
-    server {{.ID}}_{{.Address}}:{{.Port}} {{.Address}}:{{.Port}} {{ $ServiceOptions }}
+    {{ scratch.Set "weightValue" "100" -}}
+    {{ if index .Tags 0 | contains "WEIGHT:" -}}
```
Comment (ahelal) [inline etc/haproxy.ctmpl]: again what if no index 0 ?

File: etc/containerpilot.json5 (hunk 2 above)
Comment (aleerizw) [reply etc/containerpilot.json5]: Added a test in the latest commit.

File: etc/haproxy.ctmpl (hunk 3 above)
Comment (aleerizw) [reply etc/haproxy.ctmpl]: Fixed

**Comment by ahelal** [comment]
@aleerizw looks good, But can you add a test with scaling app one of the APPs to have a different weight ?

**Comment by aleerizw** [comment]
Sure, although i need to figure out how to scale an app with a different config using docker-compose.

**Comment by ahelal** [comment]
+1

**Comment by ahelal** [comment]
+1

**Pull request merged**